internal/httphandler/utils.go

```diff
@@ -36,6 +36,29 @@ func extractVerifiedQueryPayload(w http.ResponseWriter, r *http.Request, queryTy
 			DbName: params["dbname"],
 			Key:    params["key"],
 		}
+	case constants.GetDataRange:
+		limit, err := strconv.Atoi(params["limit"])
+		if err != nil {
+			utils.SendHTTPResponse(w, http.StatusBadRequest, &types.HttpResponseErr{ErrMsg: err.Error()})
+			return nil, true
+		}
+
+		if limit < 0 {
+			utils.SendHTTPResponse(
+				w,
+				http.StatusBadRequest,
+				&types.HttpResponseErr{ErrMsg: "limit value " + strconv.Itoa(limit) + " should be greater than or equal to zero"},
+			)
+			return nil, true
+		}
+
+		payload = &types.GetDataRangeQuery{
+			UserId:   querierUserID,
+			DbName:   params["dbname"],
+			StartKey: params["startkey"][1 : len(params["startkey"])-1],
+			EndKey:   params["endkey"][1 : len(params["endkey"])-1],
+			Limit:    uint32(limit),
+		}
```

Comment:
This assumes that there is always a "limit" "startkey" and "endkey". This could be made more flexible, i.e. no limit means 0; no endkey means endkey="", etc. This would also require changes to the (gorilla) matcher...

What do you think?

Reply:
Yes, in the future PRs. https://github.com/hyperledger-labs/orion-server/issues/423